tests/specs/nodeRewardsProgram.spec.js

```diff
@@ -0,0 +1,373 @@
+            done();
+        });
+
+        it('should return false when there is successfull enrollment for given public key and recipient address is not a valid enrollment address', async (done) => {
+            // Arrange:
+            spyOn(NodeRewardsProgram, 'checkEnrollmentAddress').and.returnValue(Promise.resolve(false));
+            mockFetch([{
+                recipientAddress: 'NCHESTYVD2P6P646AMY7WSNG73PCPZDUQNSD6JAK',
+                transactionHeight: 1,
+                transactionHash: 'bba40188dc1f042da959309639a884d8f6a87086cda10300d2a7c3a0e0891aua',
+                host: 'http://108.61.168.86:7890'
+            }]);
+            const publicKey = '00a30788dc1f042da959309639a884d8f6a87086cda10300d2a7c3a0e0891a4d';
+            const expectedEnrollmentStatus = false;
+
+            // Act:
+            const enrollmentStatus = await NodeRewardsProgram.checkEnrollmentStatus(publicKey);
+
+            // Assert:
+            expect(enrollmentStatus).toBe(expectedEnrollmentStatus);
+            done();
+        });
+
+        it('should return false when there is no successfull enrollment for given public key', async (done) => {
+            // Arrange:
+            mockFetch([]);
+            const publicKey = '00a30788dc1f042da959309639a884d8f6a87086cda10300d2a7c3a0e0891a4d';
+            const expectedEnrollmentStatus = false;
+
+            // Act:
+            const enrollmentStatus = await NodeRewardsProgram.checkEnrollmentStatus(publicKey);
+
+            // Assert:
+            expect(enrollmentStatus).toBe(expectedEnrollmentStatus);
+            done();
+        });
+
+        it('should throw an error when request failed', async (done) => {
+            // Arrange:
+            mockFetch(null, false);
+            const publicKey = '00a30788dc1f042da959309639a884d8f6a87086cda10300d2a7c3a0e0891a4d';
+            const expectedError = Error('failed_to_get_success_enrollments');
+
+            // Act:
+            try {
+                await NodeRewardsProgram.checkEnrollmentStatus(publicKey);
+            }
+            catch(e) {
+                // Assert:
+                expect(e.message).toBe(expectedError.message);
+                done();
+            }
+        });
+    });
+
+    describe('getCodewordHash()', () => {
+        it('should fetch to be called with valid endpoint url when publicKey passed', async (done) => {
+            // Arrange:
+            const mockedFetch = mockFetch(null);
+            const baseUrl = NODE_REWARDS_PROGRAM_API_BASE_URL;
+            const publicKey = '00a30788dc1f042da959309639a884d8f6a87086cda10300d2a7c3a0e0891a4d';
+            const expectedEndpoint = `${baseUrl}/codeword/${publicKey}`;
+
+            // Act:
+            await NodeRewardsProgram.getCodewordHash(publicKey);
+
+            // Assert:
+            expect(mockedFetch).toHaveBeenCalledWith(expectedEndpoint);
+            done()
+        });
```

Comment:
i don't think this test make much sense as standalone.
instead, check that fetch was called in following two tests.

Reply:
it depends how you define a unit. if you define a unit as something like "constructs url properly", "forwards object on success", then the separate tests make sense.

however, if you define a unit as a branch, like "handles successful fetch", "handles failed fetch", then combining makes sense. one benefit of this approach is that you will validate (in error cases) that the fetch was initiated (and that something previous didn't trigger the error). 

defining units is more of an art than a science, so you can leave current cases if you prefer.

Reply:
Thanks for the explanation, that makes sense.